llvm/lib/Target/WebAssembly/WebAssemblyISelLowering.cpp

```diff
@@ -1211,6 +1211,13 @@ WebAssemblyTargetLowering::LowerCall(CallLoweringInfo &CLI,
         }
       }
     }
+    // Similarly, we cannot tail call with byval arguments.
+    for (unsigned I = 0; I < CLI.Outs.size(); ++I) {
+      const ISD::OutputArg &Out = CLI.Outs[I];
+      if (Out.Flags.isByVal() && Out.Flags.getByValSize() != 0)
+        NoTail(
+            "WebAssembly does not support tail calling with stack arguments");
```

Comment:
This message can probably more specifically mention byval arguments.

Reply:
I was assuming that "byval argument" is not going to mean much to most developers who might see this error message.